extensions/blocks/podcast-player/podcast-player.php

```diff
@@ -214,3 +215,43 @@ function get_colors( $name, $attrs, $property ) {
 
 	return $colors;
 }
+
+/**
+ * Render the given template in server-side.
+ *
+ * @param string $name  Template name, available in `./templates` folder.
+ * @param array  $data   Template data. Optional.
+ * @param bool   $print   Render template. True as default.
+ * @return false|string HTML markup or false.
+ */
+function render( $name, $data = array(), $print = true ) {
+	if ( ! strpos( $name, '.php' ) ) {
+		$name = $name . '.php';
+	}
+
+	$template_path = dirname( __FILE__ ) . '/templates/' . $name;
+
+	if ( ! file_exists( $template_path ) ) {
+		return '';
+	}
+
+	// Optionally provided an assoc array of data to pass to template
+	// and it will be extracted into variables.
+	if ( is_array( $data ) ) {
+		// phpcs:disable WordPress.PHP.DontExtract.extract_extract
+		extract( $data );
+		// phpcs:enable WordPress.PHP.DontExtract.extract_extract
+	}
+
+	ob_start();
+	include $template_path;
+	$markup = ob_get_contents();
+	ob_end_clean();
+
+	if ( $print ) {
+		// phpcs:disable WordPress.Security.EscapeOutput.OutputNotEscaped
+		echo $markup;
```

Comment:
I wonder. Ought we to escape here as well? 🤔

Reply:
I think it's already checked for each template. Checking the checking...

Reply:
Tried to commit this change:

```git
index cd8ab23..52be68f 100644
--- a/extensions/blocks/podcast-player/templates/podcast-header.php
+++ b/extensions/blocks/podcast-player/templates/podcast-header.php
@@ -49,7 +49,7 @@ $track = ! empty( $data['tracks'] ) ? $data['tracks'][0] : array();
                id="<?php echo esc_attr( $playerId ); ?>__track-description"
                class="jetpack-podcast-player__track-description"
        >
-               <?php echo esc_attr( $track['description'] ); ?>
+               <?php echo $track['description']; ?>
        </div>
        <?php endif; ?>
 </div>
```

And got this sniffer error:

```
FILE: ...ck/extensions/blocks/podcast-player/templates/podcast-header.php
----------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
----------------------------------------------------------------------
 52 | ERROR | All output should be run through an escaping function
    |       | (see the Security sections in the WordPress Developer
    |       | Handbooks), found '$track['description']'.
    |       | (WordPress.Security.EscapeOutput.OutputNotEscaped)
----------------------------------------------------------------------

Time: 204ms; Memory: 14MB
```

So we're still safe.

Reply:
Wouldn't it break the whole markup? All brackets like `<` or `>` would get translated into HTML entities and they will not render properly. Unless there is an escaping function for full HTML but then I'm not sure I can see that as useful or secure. 

I think this one is ok, I will maybe add a comment explaining why it was disabled (output already escaped elsewhere).

Reply:
> Wouldn't it break the whole markup? All brackets like < or > would get translated into HTML entities and they will not render properly.

I think it's an answer to _...Ought we to escape here as well?_. We shouldn't escape chars here.

> I think this one is ok, I will maybe add a comment explaining why it was disabled (output already escaped elsewhere).

Yeah, agree. Updating...